src/main/kotlin/no/nav/familie/ef/sak/barn/BarnService.kt

```diff
@@ -19,47 +20,129 @@ class BarnService(
-                }
-            }
-            StønadType.OVERGANGSSTØNAD, StønadType.SKOLEPENGER -> {
-                val barnFraSøknad = finnSøknadsbarnOgMapTilBehandlingBarn(behandlingId = behandlingId) +
-                    barnSomSkalFødes.map { it.tilBehandlingBarn(behandlingId) }
-                BarnMatcher.kobleBehandlingBarnOgRegisterBarn(barnFraSøknad, grunnlagsdataBarn)
-                    .map {
-                        BehandlingBarn(
-                            id = it.behandlingBarn.id,
-                            behandlingId = behandlingId,
-                            personIdent = it.barn?.personIdent,
-                            søknadBarnId = it.behandlingBarn.søknadBarnId,
-                            navn = it.barn?.navn?.visningsnavn(),
-                            fødselTermindato = it.behandlingBarn.fødselTermindato
-                        )
-                    }
-            }
+            StønadType.BARNETILSYN -> barnForBarnetilsyn(barnSomSkalFødes, behandlingId, grunnlagsdataBarn)
+            StønadType.OVERGANGSSTØNAD, StønadType.SKOLEPENGER ->
+                kobleBarnForOvergangsstønadOgSkolepenger(
+                    behandlingId,
+                    behandlingsårsak,
+                    grunnlagsdataBarn,
+                    barnSomSkalFødes
+                )
         }
         barnRepository.insertAll(barnPåBehandlingen)
     }
 
+    /**
+     * Barnetilsyn bruker barn fra registeret, og kobler det til søknadsbarn hvis det finnes i søknaden
+     */
+    private fun barnForBarnetilsyn(
+        barnSomSkalFødes: List<BarnSomSkalFødes>,
+        behandlingId: UUID,
+        grunnlagsdataBarn: List<BarnMedIdent>
+    ): List<BehandlingBarn> {
+        feilHvis(barnSomSkalFødes.isNotEmpty()) {
+            "Kan ikke håndtere barnSomSkalFødes i barnetilsyn"
+        }
+        val barnFraSøknad = hentSøknadsbarnForBehandling(behandlingId).associate { it.fødselsnummer to it.id }
+        return grunnlagsdataBarn.map { barn ->
+            BehandlingBarn(
+                behandlingId = behandlingId,
+                søknadBarnId = barnFraSøknad[barn.personIdent],
+                personIdent = barn.personIdent,
+                navn = barn.navn.visningsnavn(),
+            )
+        }
+    }
+
+    private fun kobleBarnForOvergangsstønadOgSkolepenger(
+        behandlingId: UUID,
+        behandlingsårsak: BehandlingÅrsak,
+        grunnlagsdataBarn: List<BarnMedIdent>,
+        barnSomSkalFødes: List<BarnSomSkalFødes>
+    ): List<BehandlingBarn> {
+        feilHvis(behandlingsårsak != BehandlingÅrsak.PAPIRSØKNAD && barnSomSkalFødes.isNotEmpty()) {
+            "Kan ikke legge til terminbarn med behandlingsårsak=$behandlingsårsak"
+        }
+        return if (behandlingsårsak == BehandlingÅrsak.PAPIRSØKNAD) {
+            barnForPapirsøknad(behandlingId, barnSomSkalFødes, grunnlagsdataBarn)
+        } else {
+            kobleBehandlingBarnOgRegisterBarnTilBehandlingBarn(
```

Comment:
Kan saksbehandler opprette ny behandling basert på ettersending? 
Isåfall - da mister vi alle barna, ikke sant? Burde vi feile dersom det skjer?